aten/src/ATen/native/Embedding.cpp

```diff
@@ -68,65 +68,63 @@ Tensor embedding_sparse_backward(
 
   // check if all our grad come from padding_idx
   if (grad.numel() == 0) {
-    return at::_sparse_coo_tensor_unsafe(at::empty({1, 0}, indices_.options()),
+    return at::_sparse_coo_tensor_unsafe(at::empty({1, 0}, indices_.options().dtype(kLong)),
                                          at::empty({0, num_features}, dense_options),
                                          weight_size);
   }
 
   auto index = indices.reshape({1, -1});
   auto values = grad.reshape({-1, num_features});
-  return at::_sparse_coo_tensor_unsafe(index, values, weight_size);
+  return at::_sparse_coo_tensor_unsafe(index.to(kLong), values, weight_size);
```

Comment:
hm, so we're doing still int64 always in the backwards? are there concerns with doing int32? sparse tensor should support it.

Or the worry is that we'd need to update optimizers too?

Reply:
It doesn't matter, indices of sparse tensor will be converted to Long anyway.

Reply:
I tried to use int32 initially and found that SparseTensor or at least some SparseTensor utilities assume int64 indices - https://codebrowser.bddppq.com/pytorch/pytorch/aten/src/ATen/SparseTensorImpl.cpp.html#90